provider/resource_keycloak_identity_provider_token_exchange_scope_permission.go

```diff
@@ -0,0 +1,292 @@
+				Required:    true,
+				Description: "Ids of the clients for which a policy will be created and set on scope based token exchange permission",
+			},
+			"policy_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				ForceNew:    true,
+				Description: "Policy id that will be set on the scope based token exchange permission automatically created by enabling permissions on the reference identity provider",
+			},
+			"authorization_resource_server_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Resource server id representing the realm management client on which this permission is managed",
+			},
+			"authorization_idp_resource_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Resource id representing the identity provider, this automatically created by keycloak",
+			},
+			"authorization_token_exchange_scope_permission_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Permission id representing the Permission with scope 'Token Exchange' and the resource 'authorization_idp_resource_id', this automatically created by keycloak, the policy id will be set on this permission",
+			},
+		},
+	}
+}
+
+func setIdentityProviderTokenExchangeScopePermissionClientPolicy(keycloakClient *keycloak.KeycloakClient, realmId, providerAlias string, clients []string) error {
+	identityProviderPermissions, err := keycloakClient.GetIdentityProviderPermissions(realmId, providerAlias)
+	if err != nil {
+		return err
+	}
+
+	realmManagementClient, err := keycloakClient.GetOpenidClientByClientId(realmId, "realm-management")
+	if err != nil {
+		return err
+	}
+
+	tokenExchangeScopedPermissionId, err := identityProviderPermissions.GetTokenExchangeScopedPermissionId()
+	if err != nil {
+		return err
+	}
+
+	permission, err := keycloakClient.GetOpenidClientAuthorizationPermission(realmId, realmManagementClient.Id, tokenExchangeScopedPermissionId)
+	if err != nil {
+		return err
+	}
+
+	if len(permission.Policies) == 0 {
+		policyId, err := createClientPolicy(keycloakClient, realmId, realmManagementClient.Id, providerAlias, clients)
+		if err != nil {
+			return err
+		}
+		permission.Policies = []string{policyId}
+		return keycloakClient.UpdateOpenidClientAuthorizationPermission(permission)
+
+	} else if len(permission.Policies) == 1 {
+		openidClientAuthorizationClientPolicy, err := keycloakClient.GetOpenidClientAuthorizationClientPolicy(realmId, realmManagementClient.Id, permission.Policies[0])
+		if err != nil {
+			return err
+		}
+		openidClientAuthorizationClientPolicy.Clients = clients
+		return keycloakClient.UpdateOpenidClientAuthorizationClientPolicy(openidClientAuthorizationClientPolicy)
+
+	} else {
+		return fmt.Errorf("only supports 1 client policy is supporterd on the identity provider token exchange scope based permission, but %d were found", len(permission.Policies))
```

Comment:
```suggestion
		return fmt.Errorf("only one client policy is supported, but %d were found", len(permission.Policies))
```

cleaned up the wording a bit. the error message will already contain the resource name, so I think this is enough info for the user.